pkgs/applications/system/mgmt/default.nix

```diff
@@ -0,0 +1,68 @@
+{ augeas
+, buildGoModule
+, fetchFromGitHub
+, gotools
+, lib
+, libvirt
+, libxml2
+, nex
+, pkg-config
+, ragel
+}:
+buildGoModule rec {
+  pname = "mgmt";
+  version = "unstable-2022-10-24";
+
+  src = fetchFromGitHub {
+    owner = "purpleidea";
+    repo = pname;
+    rev = "d8820fa1855668d9e0f7a7829d9dd0d122b2c5a9";
+    hash = "sha256-jurZvEtiaTjWeDkmCJDIFlTzR5EVglfoDxkFgOilo8s=";
+  };
+
+  postPatch = ''
+    for file in `find -name Makefile -type f`; do
+      substituteInPlace $file --replace "/usr/bin/env " ""
+    done
+
+    substituteInPlace lang/types/Makefile \
+      --replace "unset GOCACHE && " ""
+    patchShebangs misc/header.sh
```

Comment:
Something is missing here

```
make: /usr/bin/env: No such file or directory
make: /usr/bin/env: No such file or directory
make: /usr/bin/env: No such file or directory
make: /usr/bin/env: No such file or directory
make: /usr/bin/env: No such file or directory
make: /usr/bin/env: No such file or directory
make: /usr/bin/env: No such file or directory
make: /usr/bin/env: No such file or directory
make: /usr/bin/env: No such file or directory
make: /usr/bin/env: No such file or directory
make: *** [Makefile:144: lang] Error 127
```

Reply:
> terribly sorry for the back and for on this one, see #208036 for full context, and I will try to ensure that everything works before requesting more reviews.

that said, it looks like we cannot use `preBuild` and something else; would you prefer that I use only `preBuild` or two other hooks, say `postPatch` and `postConfigure`?

Reply:
If we are patching this in preBuild, which is totally fine, we should add a short comment why it is not already done in postPatch